pkg/roachpb/metadata.go

```diff
@@ -925,23 +925,26 @@ func (h *GCHint) IsEmpty() bool {
 // of receiver hint (leftEmpty) and argument hint (rightEmpty).
 // Returns true if receiver state was changed.
 func (h *GCHint) Merge(rhs *GCHint, leftEmpty, rightEmpty bool) bool {
+	updated := h.AddDeleteTimestamp(rhs.MinDeleteTimestamp)
+	// NB: don't swap the operands, we need the side effect of the method call.
+	updated = h.AddDeleteTimestamp(rhs.MaxDeleteTimestamp) || updated
+
 	// If LHS or RHS has data but no LatestRangeDeleteTimestamp hint, then this
 	// side is not known to be covered by range tombstones. Correspondingly, the
 	// union of the two is not too. If so, clear the hint.
 	if (rhs.LatestRangeDeleteTimestamp.IsEmpty() && !rightEmpty) ||
 		(h.LatestRangeDeleteTimestamp.IsEmpty() && !leftEmpty) {
-		updated := h.LatestRangeDeleteTimestamp.IsSet()
+		updated = updated || h.LatestRangeDeleteTimestamp.IsSet()
 		h.LatestRangeDeleteTimestamp = hlc.Timestamp{}
-		return updated
```

Comment:
Should still `return updated` here, we don't want to call `ForwardLatestRangeDeleteTimestamp` in this case.

Need to better test this `Merge` method.

Reply:
Done.